src/components/list/ItemsList.tsx

```diff
@@ -1,63 +1,114 @@
+  renderItemsSeparator: () => Element;
+  selectionMode?: ItemSelectionMode;
   skip: number;
   count: number;
   limit: number;
   refreshing: boolean;
+  initiallySelectedItems?: T[];
 }
 
-export enum ItemsListTypes {
+export enum ItemSelectionMode {
   NONE = 'none',
   MULTI = 'multi',
   SINGLE = 'single'
 }
 
-interface State {
-  selectedIds?: Set<string>;
+interface State<T> {
+  selectedItems?: Map<string | number, T>;
 }
 
-export default class ItemsList<T extends ListItem> extends React.Component<Props<T>, State> {
+export default class ItemsList<T extends ListItem> extends React.Component<Props<T>, State<T>> {
   public static defaultProps = {
-    select: ItemsListTypes.NONE
+    selectionMode: ItemSelectionMode.NONE,
+    initiallySelectedItems: [],
+    renderItemsSeparator: () => <View style={computeStyleSheet().rowSeparator} />
   };
-  public state: State;
+  public state: State<T>;
   public props: Props<T>;
 
   public constructor(props: Props<T>) {
     super(props);
-    this.state = { selectedIds: new Set<string>() };
+    this.state = { selectedItems: new Map<string | number, T>() };
+  }
+
+  public componentDidMount() {
+    const { initiallySelectedItems } = this.props;
+    if (initiallySelectedItems) {
+      const selectedItems = new Map<string | number, T>();
+      initiallySelectedItems.forEach((item) => selectedItems.set(item?.id, item));
+      this.setState({ selectedItems });
+    }
   }
 
   public setState = (
-    state: State | ((prevState: Readonly<State>, props: Readonly<Props<T>>) => State | Pick<State, never>) | Pick<State, never>,
+    state:
+      | State<T>
+      | ((prevState: Readonly<State<T>>, props: Readonly<Props<T>>) => State<T> | Pick<State<T>, never>)
+      | Pick<State<T>, never>,
     callback?: () => void
   ) => {
     super.setState(state, callback);
   };
 
   public render() {
-    const { data, skip, count, limit, navigation, manualRefresh, refreshing, onEndReached, emptyTitle } = this.props;
-    const { selectedIds } = this.state;
+    const {
+      data,
+      skip,
+      count,
+      limit,
+      navigation,
+      manualRefresh,
+      refreshing,
+      onEndReached,
+      emptyTitle,
+      selectionMode,
+      onSelect
+    } = this.props;
+    const { selectedItems } = this.state;
+    const selectionEnabled = selectionMode !== ItemSelectionMode.NONE && onSelect;
+    const style = computeStyleSheet();
     return (
       <FlatList
         data={data}
         renderItem={({ item }) => (
-          <TouchableOpacity onPress={() => this.onSelectItem(item)}>
-            {this.props.renderItem(item, selectedIds.has(item.id as string))}
-          </TouchableOpacity>
+          <View>
+            <TouchableOpacity disabled={!selectionEnabled} onPress={() => this.onSelectItem(item)}>
+              <View style={style.rowContainer}>
+                {selectionMode === ItemSelectionMode.SINGLE && (
+                  <RadioButton.Android
```

Comment:
I just tested it and with single select, it would be better to not display at all the radio button I think as once selected, the pop-up disappear immediately.

Reply:
But in this case, we will not display the buttons when reopening

Reply:
I don't get it, I just want to remove the radio button which is useless I think.
The title is clear and invite the user to select.
Once a line is clicked, you close the pop-up (if mode = single select) and display the selected user.

Reply:
I get it, but if you remove the button, how do you indicate which item is selected when reopening the modal ?